src/authConfig.ts

```diff
@@ -16,37 +19,81 @@ declare global {
     }
 }
 
-export const createMsalConfig = (clientId: string, authority: string, redirectUri: string): Configuration => {
-    return {
-        auth: {
-            // the following two values are changed using the Environment Provider in the layout.tsx
-            clientId,
-            authority, // This is a URL (e.g. https://login.microsoftonline.com/{your tenant ID})
-            redirectUri,
+const keycloakEnabled = process.env.KEYCLOAK_ENABLED?.toLowerCase() === 'true'.toLowerCase();
+
+export const authOptions: AuthOptions = {
+    providers: [
+        ...(keycloakEnabled ? [
+            KeycloakProvider({
+                clientId: process.env.KEYCLOAK_CLIENT_ID ? process.env.KEYCLOAK_CLIENT_ID : '',
+                clientSecret: 'process.env.KEYCLOAK_CLIENT_SECRET',
```

Comment:
Why is this env variable hardcoded as a string?

Reply:
can you comment that in the code? So we don't be suprised about it in a few weeks 😄